parser/src/aspects/block_parser.rs

```diff
@@ -0,0 +1,194 @@
+use lexer::token::TokenType;
+use crate::aspects::base_parser::BaseParser;
+use crate::ast::Expr;
+use crate::parser::{Parser, ParseResult};
+use crate::ast::statement::Block;
+
+pub trait BlockParser<'a> {
+    fn block(&mut self) -> ParseResult<Expr<'a>>;
+}
+
+impl<'a> BlockParser<'a> for Parser<'a> {
+    fn block(&mut self) -> ParseResult<Expr<'a>> {
+        self.expect_token(TokenType::CurlyLeftBracket, "expected start of block expression")?;
+        let block = |exprs| Ok(Expr::Block(Block {
+            exprs
+        }));
+        let mut expressions: Vec<Expr<'a>> = Vec::new();
+        //if the block is empty
+        if self.meet_token(TokenType::CurlyRightBracket) {
+            return block(expressions);
+        }
```

Comment:
Is there any reason to have a spacial case for empty blocks, given that they seem to be handled the same way afterward?

Reply:
Well, semantically, a function declaration like `fun useless() = {}` would be correct, 
also, having empty blocks in code is useless but still correct. I think we should parse them in the first stage then we'll remove empty blocks when minimizing the AST on later stages

Reply:
What I meant is that there is exactly the same condition one line below at the beginning of the loop, so it seems redundant to do the same *is a curly right bracket* check twice. For a first pass, it should indeed not try to minimize the AST.

Reply:
you right, i needed to do this because the method would fail one some cases, let's pause this PR and wait for the ParserCursor to be completed.